### PR TITLE
Update airbyte-ci action to install dev if pipeline has been edited

### LINKED ISSUE #33418: Update ci to use dev versoin of airbyte-ci when pipeline is edited
## Problem
We cant test pipeline changes in our branch

## Solution 
Wait for a dev release to build and install that instead of latest

### PR DESCRIPTION
**Pull request by bnchrch**
## Overview
1. Updates our airbyte-ci github action to download the dev version if piplines has been modified
2. Updates our airbyte-ci release action to always run on PRs that modify pipelines

closes #33418 
## Pain points
1. Github doesn't consitently give you the sha that triggered the event: [1](https://github.com/orgs/community/discussions/25191), [2](https://github.com/orgs/community/discussions/26325)
2. The shorten sha action didnt account for this or let us override the sha it was shortening (so I removed it)
3. Waiting on a github workflow to complete depended on waiting on its check name (not filename) which felt brittle so I opted for polling instead.
4. These are github actions, the change and verify cycle is **LONG**, so lets focus on done not perfect with this one.

### CODE REVIEW

File: .github/workflows/airbyte-ci-release.yml [hunk 1]
```diff
@@ -6,6 +6,15 @@ concurrency:
...
+      - master
+    paths:
+      - "airbyte-ci/connectors/pipelines/**"
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
     paths:
       - "airbyte-ci/connectors/pipelines/**"
```
Comment (alafanechere) [inline .github/workflows/airbyte-ci-release.yml:19]: If I'm not mistaken the release will be triggered on `airbyte-ci` file changes only.
So if you make a change to a non airbyte-ci related file on a later commit the `run-dagger-pipeline/action.yml` will never resolve a binary for this commit.



I think that a dev install with pipx when the PR (not the commit) has `airbyte-ci` modified files is easier to implement.

Another option would be to always trigger this workflow (no `paths` filter) and use `tj-actions/changed-files` to only trigger a release if the branch has airbyte-ci modified files.

I'm not sure if `tj-actions/changed-files` returns the changed-files compared to the previous commit or compared to the target branch

**Review by alafanechere — CHANGES_REQUESTED** [review]

File: .github/workflows/airbyte-ci-release.yml (hunk 1 above)
Comment (bnchrch) [reply .github/workflows/airbyte-ci-release.yml:19]: > If I'm not mistaken the release will be triggered on airbyte-ci file changes only.
So interestingly enough this was the case previously when using `push.paths`. However its not the case for `pull_request.paths` + `pull_request.types=synchronize`

> I think that a dev install with pipx when the PR (not the commit) has airbyte-ci modified files is easier to implement.
I think your right. I went down this path as it would be useful to test pipeline changes using the final pyinstaller executable. As this is what the end users actually run.

But looking at this now, the steps we take to poll for new dev binaries that match your commit is brittle and more complex than we should tolerate.

TLDR: Ill update to use the pipx dev install instead

**Comment by bnchrch** [comment]
@alafanechere back up and ready for review. We no longer are using the dev executables but instead the pipx install. Much simpler.

**Review by alafanechere — APPROVED** [review]
Thanks! I confirm it works: I opened a PR based on this one on which I've modified pipeline code -> [the formatting workflow installs airbyte-ci with pipx](https://github.com/airbytehq/airbyte/actions/runs/7274291307/job/19819916449?pr=33674)
